.github/workflows/unit_tests.sh

```diff
@@ -27,6 +27,25 @@ function info() {
 
 set -ex
 
+export PATH="$HOME/.cargo/bin:$HOME/.local/bin:$PATH"
+
+function install_rust() {
+    local tmp=$(mktemp -d)
+    # The ciphers and url are borrowed from https://sh.rustup.rs
+    local ciphers=TLS_AES_128_GCM_SHA256:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_256_GCM_SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384
+    local url="https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu"
+
+    curl -V | grep -q ' OpenSSL/'
+    curl --proto =https --tlsv1.2 --ciphers $ciphers --silent --show-error --fail --location "$url/rustup-init" --output "$tmp/rustup-init"
+    curl --proto =https --tlsv1.2 --ciphers $ciphers --silent --show-error --fail --location "$url/rustup-init.sha256" --output "$tmp/rustup-init.sha256"
+    cmp <(sha256sum $tmp/rustup-init | awk '{ print $1 }') <(awk '{ print $1 }' "$tmp/rustup-init.sha256")
```

Comment:
If you use a specific stable release, it would be possible to skip rustup and hardcode the checksum if preferred.

Reply:
I think if it was necessary to install stable releases it would probably make sense to use binaries signed with the Rust signing key. Though looking at https://forge.rust-lang.org/infra/other-installation-methods.html#standalone-installers I'm not sure how to install several stable rust releases at once. The page seems to point to the latest stable release.

Assuming replacing `X.XX.X` in `https://static.rust-lang.org/dist/rust-X.XX.X-x86_64-unknown-linux-gnu.tar.gz` should do the trick, It isn't obvious how long they are supposed to be kept there though.

It would be great to figure out how to keep all those moving parts in sync automatically.

Reply:
At least stable releases are permanent AFAIK. I believe beta and maybe nightly are permanent too, but I am not sure. The per merge/try builds that are referenced by commit are like 180 days available.

Reply:
Yes, nightlies are permanent:
```
$ rustup install nightly-2015-08-01
info: syncing channel updates for 'nightly-2015-08-01-x86_64-unknown-linux-gnu'
info: downloading component 'rust'
100.3 MiB / 100.3 MiB (100 %)   4.7 MiB/s in 21s ETA:  0s
info: installing component 'rust'
info: using up to 500.0 MiB of RAM to unpack components
100.3 MiB / 100.3 MiB (100 %)  21.4 MiB/s in  4s ETA:  0s

  nightly-2015-08-01-x86_64-unknown-linux-gnu installed - rustc 1.3.0-nightly (8ce69e856 2015-07-31)
```